osu.Game.Tests/Visual/UserInterface/TestSceneChatOverlayScenarios.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        public override IReadOnlyList<Type> RequiredTypes => new[]
+        {
+            typeof(ChannelTabControl),
+            typeof(ChannelTabItem),
+            typeof(ChatOverlay),
+        };
+
+        private TestChatOverlay chatOverlay;
+
+        [Cached]
+        private ChannelManager channelManager = new ChannelManager();
+
+        private Channel channel1;
+        private Channel channel2;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            var availableChannels = (BindableList<Channel>)channelManager.AvailableChannels;
+
+            availableChannels.Add(channel1 = new Channel(new User()) { Name = "test1" });
+            availableChannels.Add(channel2 = new Channel(new User()) { Name = "test2" });
+
+            Add(chatOverlay = new TestChatOverlay
+            {
+                RelativeSizeAxes = Axes.Both,
+                Size = new Vector2(1)
+            });
+        }
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddStep("Hide chat", () => chatOverlay.Hide());
+            AddStep("Leave channels", () =>
+            {
+                channelManager.LeaveChannel(channel1);
+                channelManager.LeaveChannel(channel2);
+            });
+            AddStep("Show chat", () => chatOverlay.Show());
+        }
+
+        /// <summary>
+        /// Test that if no maps are added, the channel selector is also toggled when <see cref="ChatOverlay"/> is toggled.
+        /// Also check that both are properly closed when toggling again.
+        /// </summary>
+        [Test]
+        public void TestToggleChatWithNoChannelsJoined()
+        {
+            AddAssert("Channel selection overlay hidden", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
+            //TODO: Change this to check whether or not the chat overlay was shown once https://github.com/ppy/osu/issues/5161 is fixed
+            AddAssert("Chat overlay was shown", () => chatOverlay.State.Value == Visibility.Visible);
+            AddStep("Close chat overlay", () => chatOverlay.Hide());
+            AddAssert("Channel selection overlay was hidden", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
+            AddAssert("Chat overlay was hidden", () => chatOverlay.State.Value == Visibility.Hidden);
+        }
+
+        [Test]
+        public void TestToggleChatWithChannelJoined()
+        {
+            AddStep("Join channel 1", () =>
+            {
+                channelManager.JoinChannel(channel1);
+                // TODO: Temporarily here to circumvent https://github.com/ppy/osu/issues/5152. Remove once fixed.
+                channelManager.OpenChannel(channel1.Name);
+            });
+            AddStep("Close chat overlay", () => chatOverlay.Hide());
+            AddAssert("Channel selection overlay was hidden", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
```

Comment:
The overlay is still open from setup, though perhaps it would be better to not have setup open the overlay by default

Reply:
the hide here is checking the selection overlay state, so the hide on *chatOverlay* should not be required for this assert to become true, no?

Reply:
Joining a channel shouldn't hide the selection overlay though (which is currently [kinda broken](https://github.com/ppy/osu/issues/5157)). This test is really mostly checking that the chat overlay doesn't show the selector if there's a channel already joined when toggled.

A separate test will be made to test that scenario when the PR for that issue is made.

Reply:
Yes, which is why I'm saying the hide step should not be there (especially if you are testing what you just said).

Reply:
If we don't hide the chat overlay, how do we test the show step from a fully invisible state (since setup makes it visible)?

Reply:
I"m lost

Reply:
Let me clarify what this test is testing. 

Its basically the opposite case of the test that was deleted above. The flow looks like this:

 - A channel is joined.
 - Hide chat
 - Show chat
 - Channel selector should be hidden.

The additional asserts are more for safety.

Reply:
Are you implying the channel selector should not be asserted before the chat is hidden? You understand I'm proposing to change the order of these two lines because the selector should *logically* be hidden *before*t he overlay is hidden.

Reply:
Ah, I see where the misunderstanding occured. That `OpenChannel` is not supposed to be in that test (its there to circumvent another issue for now). If it was not, we'd usually expect the channel selector to still be open in that scenario, since only a join channel was performed.

Reply:
Just remove these tests and add them in the future PR. Keep tests in here which apply to this PR.

Reply:
the rest of the asserts are kinda still relevant though, I guess I'll just remove part of them for now